src/receiver-ui.py

```diff
@@ -0,0 +1,119 @@
+import ipaddress, subprocess
+from tkinter import *
+from tkinter import filedialog, messagebox
+from receiver import Receiver
+
+
+def register():
+    receiver.display_name = display_name_entry.get()
+    receiver.serial_number = serial_number_entry.get()
+    return_message = receiver.register()
+    if return_message == "Decoder already registered!":
+        messagebox.showerror("Error", return_message)
+    else:
+        messagebox.showinfo("Info", return_message)
+
+
+def receive():
+    ip = ip_entry.get()
+    port = port_entry.get()
+    if not is_valid_ip(ip):
+        messagebox.showerror("Error", "Invalid IP address.")
+        return
```

Comment:
This will reject `localhost` as an IP address, though ffmpeg will accept that as a valid IP

Reply:
Great point - fixed